core/src/main/java/de/danoeh/antennapod/core/preferences/UserPreferences.java

```diff
@@ -1084,4 +1085,24 @@ public static void resetUsageCountingDate() {
     public static void unsetUsageCountingDate() {
         setUsageCountingDateMillis(-1);
     }
+
+    public static Set<String> getTagFilterIds() {
+        return prefs.getStringSet(PREF_TAG_FILTER, new HashSet<>());
+    }
+
+    public static void addTagFilterId(long tagFilterId) {
+        Set<String> tagFilterIds = prefs.getStringSet(PREF_TAG_FILTER, new HashSet<>());
```

Comment:
This stores which tags are selected on the subscriptions screen, right? So this would only be needed on the subscriptions screen. The UserPreferences class is already monstrous, so please move the code directly to the subscriptions screen.

Reply:
But this allows the the selected tags to be persisted. If removed, the user would have to re select tags. I think it is a useful feature to have. If the User preferences is too large, I wouldn't mind helping to refactor it.

Reply:
My idea was not to not store it. It was to move the storage code to the only fragment that uses it - just like how the number of columns is stored directly in the subscription fragment. This also decouples the different components of the app from each other.